pyAudioAnalysis/audioSegmentation2.py

```diff
@@ -465,15 +465,21 @@ def save_hmm(hmm_model_name, model, classes, mid_window, mid_step):
         cpickle.dump(mid_step, f_handle, protocol=cpickle.HIGHEST_PROTOCOL)
 
 
-def hmm_segmentation(audio_file, hmm_model_name, plot_results=False,
-                     gt_file=""):
+def hmm_segmentation(audio_file, hmm_model_name, plot_results=False, gt_file="", mf_object=None):
     sampling_rate, signal = audioBasicIO.read_audio_file(audio_file)
 
-    with open(hmm_model_name, "rb") as f_handle:
-        hmm = cpickle.load(f_handle)
-        class_names = cpickle.load(f_handle)
-        mid_window = cpickle.load(f_handle)
-        mid_step = cpickle.load(f_handle)
+    if mf_object is not None:
+        mf_object.seek(0)
+        hmm = cpickle.load(mf_object)
+        class_names = cpickle.load(mf_object)
+        mid_window = cpickle.load(mf_object)
+        mid_step = cpickle.load(mf_object)
+    else:
+        with open(hmm_model_name, "rb") as f_handle:
+            hmm = cpickle.load(f_handle)
+            class_names = cpickle.load(f_handle)
+            mid_window = cpickle.load(f_handle)
+            mid_step = cpickle.load(f_handle)
```

Comment:
Suggestion (applies both to this piece of code and there's similar one down the diff):

maybe have a "private" helper func to load stuff?

Like:

```
def load_SOMETHING(mf_object):
  hmm = cpickle.load(...)
  ...
  mid_step = cpickle.load(mf_object)

def hmm_segmentation(...):
  if mf_object is not None:
    hmm, class_names, ... = load_SOMETHING(mf_object)
  else:
    with open(...) as mf_object:
      hmm, class_names, ... = load_SOMETHING(mf_object)
```

Like, copy-paste can be fine in this place (just 4 straightforward cpickle.load-s), but down the diff - there's much more to load.

And maintaining a duplicated code is error-prone (update one place, forget about another - that stuff).